<a href="https://colab.research.google.com/github/AhmedAnasHaouari/project_2/blob/main/Project_2_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

  # **PROJECT 2**:

- Haouari Ahmed Anas

- 09/11/2023

# Imports :

In [51]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [52]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.impute import SimpleImputer
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.dummy import DummyRegressor
from sklearn.compose import ColumnTransformer
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
from sklearn.preprocessing import OneHotEncoder, StandardScaler, OrdinalEncoder
from sklearn.compose import make_column_selector, make_column_transformer
from sklearn.pipeline import make_pipeline
from sklearn import set_config
set_config(transform_output='pandas')

# Loading Data :

In [53]:
file1 = "/content/drive/MyDrive/datasets/adult.csv"
file2 = '/content/drive/MyDrive/datasets/healthcare-dataset-stroke-data.csv'
df_inc = pd.read_csv(file1)
df_str = pd.read_csv(file2)
df_inc.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [54]:
df_inc.shape

(48842, 15)

In [55]:
df_str.head()

id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [56]:
df_str.shape

(5110, 12)

 - Sources :
  - Dataset 1: wenruliu. (2016). Adult income dataset. Retrieved from https://www.kaggle.com/datasets/wenruliu/adult-income-dataset
  
  - Dataset 2: fedesoriano. (2020). Stroke Prediction Dataset. Retrieved from https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset.
  
  

- Brief description of data :

  - Dataset 1 :  An individual’s annual income results from various factors. Intuitively, it is influenced by the individual’s education level, age, gender, occupation, and etc. The detailed description on the dataset can be found in the original UCI documentation
  http://www.cs.toronto.edu/~delve/data/adult/adultDetail.html

  - Dataset 2 :  According to the World Health Organization (WHO) stroke is the 2nd leading cause of death globally, responsible for approximately 11% of total deaths. This dataset is used to predict whether a patient is likely to get stroke based on the input parameters like gender, age, various diseases, and smoking status. Each row in the data provides relavant information about the patient.

- Targets :

  - Dataset 1 : Income.
  - Dataset 2 : Stroke.

- Rows :

  - Dataset 1 : 48842 person.
  - Dataset 2 : 5110 patient.


- Type of problem :

  - Dataset 1 : Classification.
  - Dataset 2 : Classification.

- Features :

  - Dataset 1 : 15 features.
  - Dataset 2 : 12 features.

# Exploring Data :

- Dataset 1 :

In [ ]:
df_inc.isna().sum()

In [ ]:
df_inc.duplicated().sum()

In [63]:
df_inc.drop_duplicates(inplace = True)

In [ ]:
df_inc.describe().round(2)

In [ ]:
df_inc.info()

In [ ]:
df_inc["native-country"].value_counts()

In [ ]:
df_inc["workclass"].value_counts()

In [ ]:
df_inc["education"].value_counts()

In [ ]:
df_inc["occupation"].value_counts()

In [ ]:
df_inc["relationship"].value_counts()

In [ ]:
df_inc["race"].value_counts()

In [ ]:
df_inc["gender"].value_counts()

In [ ]:
df_inc["income"].value_counts()

- Dataset 2 :

In [ ]:
df_str.isna().sum()

In [ ]:
df_str.duplicated().sum()

In [ ]:
df_str.describe().round(2)

In [ ]:
df_str.info()

In [ ]:
df_str["ever_married"].value_counts()

In [ ]:
df_str["Residence_type"].value_counts()

In [ ]:
df_str["smoking_status"].value_counts()

In [ ]:
df_str["gender"].value_counts()

In [105]:
other = df_str['gender'] == 'Other'
df_str = df_str[~other]

In [ ]:
df_str["work_type"].value_counts()

# Challenges :

- There are no challenges in exploring, cleaning or modeling those datasets.
- For the second dataset (df_str) there are 201 missing values in the bmi column. We can impute those missing values with the mean in the next steps.